runtime/src/main/java/org/corfudb/runtime/collections/WrappedRocksIterator.java

```diff
@@ -0,0 +1,143 @@
+package org.corfudb.runtime.collections;
+
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+import org.rocksdb.RocksIteratorInterface;
+
+/**
+ *
+ * A wrapper class that makes a access to the RocksIterator safe: the rocks library does some checks but they
+ * are enforced via assert, but assert checking is disabled on many jvms by default, hence the explicit checking.
+ *
+ * Created by Maithem on 1/24/20.
+ */
+public class WrappedRocksIterator implements RocksIteratorInterface {
+
+    private final RocksIterator iterator;
+    public WrappedRocksIterator(RocksIterator iterator) {
+        this.iterator = iterator;
+    }
+
+    /**
+     * A helper method that is called before accessing the iterator.
+     * This is necessary to make sure that access happens on an owned handle.
+     */
+    private void checkHandle() {
+        if (!iterator.isOwningHandle()) {
+            throw new IllegalStateException("RocksIterator handle is not valid");
+        }
```

Comment:
This message should be more descriptive. The only was for the handle to be invalid is either if:

1) There is some other thread trying to access the iterator.
2) The iterator has been previously closed.

The message should reflect the above conditions.